### PR TITLE
multi-label classification example: CelebA image tagging

### PR DESCRIPTION
**Pull request by aditya1503**
Adds model training code for CelebA to accompany cleanlab's main multi-label classification tutorial.

### CODE REVIEW

**Comment by cgnorthcutt** [comment]
Remember to add a 3 lines of code snippet at the top if possible that highlights example of the main methods and how to use them (doesn't need to be runnable) -- in a markdown block.

All tutorials and docstrings should have this eventually.

File: multi_label_classification/training.ipynb [hunk 1]
```diff
@@ -0,0 +1,739 @@
+{
```
Comment (jwmueller) [inline multi_label_classification/training.ipynb]: Let's call this folder: 
`multilabel_classification/`

Comment (jwmueller) [reply multi_label_classification/training.ipynb]: Let's call this notebook: 
`pytorch_network_training.ipynb`

File: multi_label_classification/tutorial-predprobs.ipynb [hunk 2]
```diff
@@ -0,0 +1,249 @@
+{
```
Comment (jwmueller) [inline multi_label_classification/tutorial-predprobs.ipynb]: Move this to main cleanlab repo. There let's name this file:
`multilabel_classification.ipynb`

Comment (jwmueller) [reply multi_label_classification/tutorial-predprobs.ipynb]: The tutorial should start with a markdown cell with the name of the tutorial:

\# Find label issues in Multi-Label Classification Data

File: multilabel_classification/pytorch_network_training.ipynb [hunk 3]
```diff
@@ -0,0 +1,1064 @@
...
+    "from imp import reload\n",
+    "from collections import defaultdict\n",
+    "from types import SimpleNamespace\n",
+    "\n",
+    "\n",
+    "import numpy as np\n",
+    "from sklearn.metrics import accuracy_score\n",
+    "from collections import Counter\n",
+    "from timm.optim import create_optimizer\n",
+    "import pickle"
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb]:
```suggestion
    "import pickle",
    "from imp import reload\n",
    "from collections import defaultdict, Counter\n",
    "from types import SimpleNamespace\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from sklearn.metrics import balanced_accuracy_score\n",
    "from sklearn.metrics import accuracy_score\n",
    "import torch\n",
    "from torch import nn\n",
    "import torch.utils.data as data\n",
    "import timm\n",
    "from timm.data.loader import create_loader\n",
    "from timm.optim import create_optimizer"
```

File: multilabel_classification/pytorch_network_training.ipynb [hunk 4]
```diff
@@ -0,0 +1,1064 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from timm.data import (\n",
+    "    resolve_data_config,\n",
+    ")\n",
+    "from timm.models import create_model"
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb]: let's move all imports to top of notebook

File: multilabel_classification/pytorch_network_training.ipynb [hunk 5]
```diff
@@ -0,0 +1,1064 @@
...
+    "!pwd"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_csv(\"img_align_celeba/all.csv\")\n",
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb]: need to get this file from somewhere?  This example should be programmatically runnable

File: multilabel_classification/pytorch_network_training.ipynb [hunk 6]
```diff
@@ -0,0 +1,1064 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.metrics import balanced_accuracy_score\n",
+    "from sklearn.metrics import accuracy_score"
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb]: unnecessary

File: multilabel_classification/pytorch_network_training.ipynb [hunk 7]
```diff
@@ -0,0 +1,1064 @@
+{
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb:1]: First cell should always be markdown with a title and short description of what the notebook does.

In this case, title can be:  Training a neural network for multi-label classification on the Celeb-A dataset

File: multilabel_classification/pytorch_network_training.ipynb [hunk 8]
```diff
@@ -0,0 +1,1064 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb]: Most people will view this notebook through github's viewer:
https://github.com/cleanlab/examples/blob/fbb88a4055cb7f942520dc59409551a559de71dc/multilabel_classification/pytorch_network_training.ipynb

You can see this output cell is huge and takes forever to scroll through, so it would be good to print less output by default (you can add `verbosity` arg to your model for example and reduce the verbosity level to suppress constant printing of the loss)

File: multilabel_classification/pytorch_network_training.ipynb [hunk 9]
```diff
@@ -0,0 +1,1064 @@
...
+    {
+     "data": {
+      "text/plain": [
+       "<module 'os' from '/usr/lib/python3.8/os.py'>"
+      ]
+     },
+     "execution_count": 20,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb]: get rid of this?  not sure what it is doing here

File: multilabel_classification/pytorch_network_training.ipynb [hunk 10]
```diff
@@ -0,0 +1,1064 @@
...
+      ]
+     },
+     "execution_count": 38,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pickle.dump(data,open(\"data.p\",'wb'))"
+   ]
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb]: these can all be one cell that is used to save all the results.

File: multilabel_classification/pytorch_network_training.ipynb [hunk 11]
```diff
@@ -0,0 +1,1064 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb]: delete this empty cell

File: multilabel_classification/pytorch_network_training.ipynb [hunk 12]
```diff
@@ -0,0 +1,1064 @@
...
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.5"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 4
+}
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb:554]: Don't worry about polishing the text too much, I'll do a rewrite of the text for you to make the notebook as clear as possible

File: multilabel_classification/pytorch_network_training.ipynb (hunk 5 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb]: Yes the wget scripts will be added here,

File: multilabel_classification/pytorch_network_training.ipynb [hunk 13]
```diff
@@ -0,0 +1,1064 @@
...
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.5"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 4
```
Comment (jwmueller) [inline multilabel_classification/pytorch_network_training.ipynb:553]: You'll want to add a requirements.txt file in this folder that lists the dependencies (and specific versions you used). See an example here:
https://github.com/cleanlab/examples/blob/fbb88a4055cb7f942520dc59409551a559de71dc/huggingface_keras_imdb/requirements.txt

If possible, try to use versions that are compatible with our main requirements.txt file for the whole examples/ repo:
https://github.com/cleanlab/examples/blob/fbb88a4055cb7f942520dc59409551a559de71dc/requirements.txt

File: multi_label_classification/tutorial-predprobs.ipynb (hunk 2 above)
Comment (aditya1503) [reply multi_label_classification/tutorial-predprobs.ipynb]: Added to first cell

File: multilabel_classification/pytorch_network_training.ipynb (hunk 4 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb]: done : https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e

File: multilabel_classification/pytorch_network_training.ipynb (hunk 5 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb]: https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e

File: multilabel_classification/pytorch_network_training.ipynb (hunk 6 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb]: Done: https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e

File: multilabel_classification/pytorch_network_training.ipynb (hunk 7 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb:1]: Done: https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e

File: multilabel_classification/pytorch_network_training.ipynb (hunk 8 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb]: Done: https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e

File: multilabel_classification/pytorch_network_training.ipynb (hunk 9 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb]: Done: https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e

File: multilabel_classification/pytorch_network_training.ipynb (hunk 10 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb]: Done: https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e

File: multilabel_classification/pytorch_network_training.ipynb (hunk 13 above)
Comment (aditya1503) [reply multilabel_classification/pytorch_network_training.ipynb:553]: Done: https://github.com/cleanlab/examples/pull/27/commits/28f62b8528fdc7aec2fd87bca2d5da8e142b7b1e

File: multilabel_classification/requirements.txt [hunk 14]
```diff
@@ -0,0 +1,7 @@
+numpy==1.22.0
+scikit-learn==1.0.2
+pillow==9.0.1
+torch==1.12.1+cu113
+timm==0.6.7
+pandas==1.5.0
+torchvision==0.13.1+cu102
```
Comment (jwmueller) [inline multilabel_classification/requirements.txt]:
```suggestion
```

Comment (jwmueller) [reply multilabel_classification/requirements.txt]: I don't see where you need torchvision?

Comment (aditya1503) [reply multilabel_classification/requirements.txt:7]: img = torchvision.io.read_image(path).swapaxes(0, 2).swapaxes(0, 1)

jpeg is not read by PIL/ other libraries, I could add just torch vision without CUDA, 
I'll also see if there are other libraries that can easily display jpeg images

Comment (aditya1503) [reply multilabel_classification/requirements.txt]: This was me trying to use torch vision to download CelebA dataset, I'll keep this resolve open, once I figure this out I'll close it

**Review by jwmueller — APPROVED** [review]
Made another final pass of text/code edits, so please review these and give me thumbs up before I merge this.

TODOs for follow-up PR after this is merged (and the official links to these notebooks are available):
- [x] make sure the links in these notebooks to one another are correct
- [x] make sure the link to these notebooks in the tutorial notebook is correct
- [x] add notebooks + short description to table of contents in main readme of this repo

**Comment by aditya1503** [comment]
> Made another final pass of text/code edits, so please review these and give me thumbs up before I merge this.
> 
> TODOs for follow-up PR after this is merged (and the official links to these notebooks are available):
> 
> * [ ]  make sure the links in these notebooks to one another are correct
> * [ ]  make sure the link to these notebooks in the tutorial notebook is correct
> * [ ]  add notebooks + short description to table of contents in main readme of this repo

OK

**Comment by aditya1503** [comment]
> Made another final pass of text/code edits, so please review these and give me thumbs up before I merge this.
> 
> TODOs for follow-up PR after this is merged (and the official links to these notebooks are available):
> 
> * [x]  make sure the links in these notebooks to one another are correct
> * [x]  make sure the link to these notebooks in the tutorial notebook is correct
> * [x]  add notebooks + short description to table of contents in main readme of this repo

LGTM! 

minor change: [Kaggle](https://www.kaggle.com/datasets/jessicali9530/celeba-dataset) link is unofficial, so I replaced them with the original dataset [link](https://mmlab.ie.cuhk.edu.hk/projects/CelebA.html)
I've added the notebooks to README.md
